c/shared/source/tx_shared/tx_shared.c

```diff
@@ -405,6 +405,31 @@ static size_t stm_read(ctlStreamCallbacks *cb, void *stream, char **ptr) {
     return 0; /* Suppress compiler warning */
 }
 
+static size_t stm_xml_read(ctlStreamCallbacks *cb, void *stream, char **ptr, xmlDocPtr *doc){
+    int res, size = BUFSIZ;
+    char chars[BUFSIZ];
```

Comment:
Confused about `chars` and `s->buf`. I don't see where we use the former and therefore am worried about the relation between `size` (which we set here`) and `s->buf` (which presumably comes in with `stream`?).